pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codesize/xml/NcssCount.xml

```diff
@@ -184,7 +184,7 @@ public class Foo {
     </test-code>
 
     <code-fragment id="complicated-method"><![CDATA[
-public class Foo {
+public class Boo {
```

Comment:
Was the name "Foo" a problem? We have a lot of sample / test code, which define classes named "Foo". If this was a problem, we might have more problems, I wanted to say :)

Reply:
Yeah actually it was not a problem before because of the visitor, which overwrote the Memoizer of classes and methods with a new one, so it was effectively reset. Now we don't have a visitor, we just create a memoizer if we need it and it doesn't exist. As the project memoizer is not reset between each test run, we reuse the results of the previous test, unless the class or methods are named differently. We could add
```java
   @Override
    protected Rule reinitializeRule(Rule rule) {
        MetricsHook.reset();
        return rule;
    }
```
to CodeSizeRulesTest as a temporary measure, just like for AllMetricsTest. Especially so since I found that at least one other test for NcssCount was corrupted in the same manner. 

Ideally we would reset the memoizers more elegantly, like what you said about the LanguageHandlers